compiler/rustc_abi/src/lib.rs

```diff
@@ -88,6 +88,24 @@ pub struct ReprOptions {
 }
 
 impl ReprOptions {
+    pub fn new(int: Option<IntegerType>, align: Option<Align>, pack: Option<Align>) -> Self {
+        Self { int, align, pack, ..ReprOptions::default() }
+    }
+
+    pub fn with_flags(mut self, flags: ReprFlags) -> Self {
+        self.flags = flags;
+        self
+    }
+
+    #[cfg(feature = "randomize")]
```

Comment:
an alternative to this would also be to just remove the feature and make the seed a `u64` instead. that doesn't really make the code much worse and should fit rust-analyzers needs

Reply:
I went this way. The feature couldn't be removed as it will enable the `rand` dependency which r-a had some problem with it IIRC. So now we are in a non ideal state were the `field_shuffle_seed` exists but has no effect without the `randomize` feature.